### PR TITLE
Fix for NVIDIAClient output to redact the api_key

### PR DESCRIPTION
**Pull request by raspawar**
Fix for the client as the output includes the Bearer key, unredacted.

### CODE REVIEW

File: libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py [hunk 1]
```diff
@@ -195,12 +187,20 @@ def _post(
         """Method for posting to the AI Foundation Model Function API."""
         self.last_inputs = {
             "url": invoke_url,
-            "headers": self.headers["call"],
+            # "headers": self.headers["call"],
             "json": self.payload_fn(payload),
             "stream": False,
         }
+        headers = {
```
Comment (mattf) [inline libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py:194]: how about leaving `last_inputs` the same and passing call `session.post(**self._add_bearer(self.last_inputs))`

`_add_bearer` would do the appropriate checks for api_key and replacement.

this approach would reduce duplication across `session.post` call sites and allow `last_inputs` to include the `"Authorization": "Bearer {api_key}"` information.

**Review by mattf — COMMENTED** [review]
please add either a unit or integration test for this.

File: libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py (hunk 1 above)
Comment (raspawar) [reply libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py:194]: If I process in last_input then the authorization shows the api_key as it is changing inplace

File: libs/ai-endpoints/tests/unit_tests/test_api_key.py [hunk 2]
```diff
@@ -43,3 +44,13 @@ def get_api_key(instance: Any) -> str:
         assert get_api_key(public_class(nvidia_api_key="PARAM")) == "PARAM"
         assert get_api_key(public_class(api_key="PARAM")) == "PARAM"
         assert get_api_key(public_class(api_key="LOW", nvidia_api_key="HIGH")) == "HIGH"
+
```
Comment (mattf) [inline libs/ai-endpoints/tests/unit_tests/test_api_key.py:47]: we also need a test where a session.post is made and the last_input is validated to not include the secret key

**Pull request merged**